### ISSUE TITLE
Commit title displayed instead of issue title when referenced in a commit

### ISSUE THREAD

**Issue by Raul6469** [desktop/desktop#2257]
### Description
When you reference an issue number in the commit name, Desktop correctly suggests the name of the related issue, as shown in the illustration below.

![image](https://user-images.githubusercontent.com/24607388/28267418-45fbe940-6afa-11e7-8a7c-e5f51daf9ea5.png)

But, once you commit your changes, and you hover the issue number in the history tab, the commit name is displayed, instead of the issue name, as shown in this screenshot :

![image](https://user-images.githubusercontent.com/24607388/28267592-000f5c18-6afb-11e7-9b00-438736c29343.png)

### Version
**GitHub Desktop version:** 0.6.2
**OS version:** Windows 10.0.15063

### Steps to reproduce
1. Reference an issue number in your commit title
2. Commit the changes
3. In the history tab, hover the issue number in the commit title

**Expected behaviour:** the related issue title should be displayed

**Actual behaviour:** the commit title is displayed.

**Reproduces how often:** Always

**Comment by joshaber** [comment]
Nice find @Raul6469! Looks like we need a more specific tooltip for the link element.

Solving this will actually be a bit tricky. Our issues cache is specifically for use in issue autocompletion, so it doesn't include closed issues. We'd need to start caching closed issues too, or fetch them lazily as needed.

**Comment by joshaber** [comment]
Right, sorry, I was talking about how the actual implementation would work. The feature is the same regardless of whether the issue is open or closed.

**Comment by shiftkey** [comment]
Two other things to note:

 - the tooltip isn't related to the link -  you can hover over the commit title and see the commit title (yay!).

<img width="532" src="https://user-images.githubusercontent.com/359239/28530504-74fc97ec-706a-11e7-9c4d-4998489e362c.png">

 - I *think* we should be fine whether the linked item is an issue or a PR, but we'll probably need to start caching PRs anyway for other features down the track

**Comment by Raul6469** [comment]
So this is not a bug, right ? The tooltip was never meant to display the related issue name, so I think you can remove the bug label from this issue, it is more likely a feature request

**Comment by shiftkey** [comment]
@Raul6469 yeah, the context menu isn't looking up the issue information currently. I'll come back to this tomorrow and see where it fits in around other tasks on the roadmap.
